src/serve/hosting.ts

```diff
@@ -9,22 +10,59 @@ import { normalizedHostingConfigs } from "../hosting/normalizedHostingConfigs";
 import * as utils from "../utils";
 import cloudRunProxy from "../hosting/cloudRunProxy";
 import functionsProxy from "../hosting/functionsProxy";
+import { NextFunction, Request, Response } from "express";
+import { Writable } from "stream";
 
 const MAX_PORT_ATTEMPTS = 10;
 let attempts = 0;
 let server: any; // eslint-disable-line @typescript-eslint/no-explicit-any
 
+const logData = {
+  metadata: {
+    emulator: {
+      name: "hosting",
+    },
+  },
+};
+
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 function startServer(options: any, config: any, port: number, init: TemplateServerResponse): void {
+  const firebaseMiddleware = initMiddleware(init);
+
+  const morganStream = new Writable();
+  morganStream._write = (
+    chunk: any,
+    encoding: string,
+    callback: (error?: Error | null) => void
+  ) => {
+    if (chunk instanceof Buffer) {
+      utils.logLabeledBullet("hosting", chunk.toString().trim(), "info", logData);
+    }
+
+    callback();
+  };
+
+  const morganMiddleware = morgan("combined", {
+    stream: morganStream,
+  });
+
   server = superstatic({
-    debug: true,
+    debug: false,
     port: port,
     host: options.host,
     config: config,
     cwd: detectProjectRoot(options.cwd),
     stack: "strict",
     before: {
-      files: initMiddleware(init),
+      middleware: (req: Request, res: Response, next: NextFunction) => {
```

Comment:
This key was `files` before and now it's `middleware`.  Is one a more general version of the other?

Reply:
the keys are arbitrary before it dealt with the `/__/*` files, now it handles all middleware